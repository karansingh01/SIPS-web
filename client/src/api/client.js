"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const client_1 = require("@apollo/client");
const apollo_link_rest_1 = require("apollo-link-rest");
// Set rest-link endpoint
const restLink = new apollo_link_rest_1.RestLink({ uri: 'https://www.thecocktaildb.com/api/json/v1/1/' });
// Create apollo client
exports.client = new client_1.ApolloClient({
    cache: new client_1.InMemoryCache(),
    link: restLink,
});
