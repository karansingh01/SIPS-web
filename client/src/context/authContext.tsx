import jwtDecode from "jwt-decode";
import React, { useReducer, createContext } from "react";

const initialState = {
    user: null
}

if (localStorage.getItem("token")) {
    const decodedToken = jwtDecode(localStorage.getItem("token")!);

    // checks if token is expired, compares current time to token expiry time
    //@ts-ignore as it's not a string
    if (decodedToken.exp * 1000 < Date.now()) {
        localStorage.removeItem("token");
    } else {
        //@ts-ignore as I know it won't be null
        initialState.user = decodedToken;
    }
}


// create context for auth
const AuthContext = createContext({
    user: null,
    login: (userData: any) => {},
    logout: () => {}
});

// in the case of logging in, we want to return the current state, and change the state of the user to the payload of the action
// in the case of logging out, we want to return the current state, and change the state of the user to null
function authReducer(state: any, action: any) {
    switch (action.type) {
        case "LOGIN":
            return {
                ...state,
                user: action.payload
            }
        case "LOGOUT":
            return {
                ...state,
                user: null
            }
        default:
            return state;
    }
}

// react hook that has a state and a dispatch that comes from useReducer
// dispatch lets us dispatch actions to the reducer
function AuthProvider(props: any) {
    const [state, dispatch] = useReducer(authReducer, initialState);

    // when this login function is called it means we have what we need from the server to log in
    const login = (userData: any) => {
        localStorage.setItem("token", userData.token);
        // dispatch the action to the reducer, user is then set to the payload which is userData
        dispatch({
            type: "LOGIN",
            payload: userData
        });
    }

    // logs out by dispatching the logout action and removing token
    function logout() {
        localStorage.removeItem("token");
        dispatch({ type: "LOGOUT" });
    }

    // returns the provider, which is the context, and the value is the state and the login and logout functions
    // we pass in the props because they are coming from the props that are intially passed into the AuthProvider
    return (
        <AuthContext.Provider
            value={{ user: state.user, login, logout }}
            {...props}
        />
    )
}

export { AuthContext, AuthProvider };