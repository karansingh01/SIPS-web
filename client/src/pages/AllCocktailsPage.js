"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const react_2 = require("@chakra-ui/react");
const Header_1 = __importDefault(require("../components/Header"));
const SearchBar_1 = __importDefault(require("../components/SearchBar"));
const react_3 = require("react");
const DummyData_1 = __importDefault(require("../DummyData"));
const CocktailCardsDisplay_1 = __importDefault(require("../components/CocktailCardsDisplay"));
/**
 * search filter method
 * @param cocktails all cocktails
 * @param query string entered in search field
 * @returns all cocktails containing the search param
 */
const filterCocktails = (cocktails, query) => {
    if (query === "") {
        return cocktails;
    }
    else {
        const filtered = cocktails.filter((cocktail) => {
            return Object.values(cocktail.name)
                .join("")
                .toLowerCase()
                .includes(query.toLowerCase());
        });
        return filtered;
    }
};
const AllCocktailsPage = () => {
    const [query, setQuery] = (0, react_3.useState)("");
    const cocktails = DummyData_1.default;
    const filteredCocktails = filterCocktails(cocktails, query);
    return (<react_2.Flex flexDirection="column">
      <Header_1.default />
      <react_2.Grid transform="translate(0px, -150px)" margin={"30px"} rounded={20} templateRows="repeat(1fr)" templateColumns="repeat(1fr)" gap={4} mt={5}>
        <react_2.GridItem colSpan={4}>
          <SearchBar_1.default q={query} setQuery={setQuery}/>
        </react_2.GridItem>
        <react_2.GridItem colSpan={4}>
          <CocktailCardsDisplay_1.default cocktails={filteredCocktails}/>
        </react_2.GridItem>
      </react_2.Grid>
    </react_2.Flex>);
};
exports.default = AllCocktailsPage;
