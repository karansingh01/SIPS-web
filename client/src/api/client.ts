import {ApolloClient, InMemoryCache} from '@apollo/client';
import { RestLink } from 'apollo-link-rest';

// Set rest-link endpoint
const restLink = new RestLink(
    { uri: 'https://www.thecocktaildb.com/api/json/v1/1/' }
);

// Create apollo client
export const client = new ApolloClient({
    cache: new InMemoryCache(),
    link: restLink,
});


