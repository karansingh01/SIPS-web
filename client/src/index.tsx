import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import { ApolloClient, InMemoryCache, ApolloProvider, gql } from '@apollo/client';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);

console.log("he eehe");

const client = new ApolloClient({
  uri: 'www.thecocktaildb.com/api/json/v1/1/search.php?s=margarita',   // her mÃ¥ iv hente API?
  cache: new InMemoryCache(),  
});
console.log(client);

/* 
client
  .query({
    query: gql`
      query getDrinks {
        idDrink: 11007{
          strDrink
        }
      }
    `,
  })
  .then((result) => console.log(result + "asdfghjkjhgf"));

  console.log(client.link);
 */


  root.render(
    <ApolloProvider client={client}>
      <App/>
    </ApolloProvider>,
  );
