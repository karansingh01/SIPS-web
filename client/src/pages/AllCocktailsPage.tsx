import React from "react";
import { Flex, Grid, GridItem } from "@chakra-ui/react";
import { client } from "../api/client";
import Header from "../components/Header";
import SearchBar from "../components/SearchBar";
import { useState, useEffect } from "react";
import dummyCocktails from "../DummyData";
import CocktailCardsDisplay from "../components/CocktailCardsDisplay";
import { VodkaDrinksQuery } from "../api/graphql/vodkaDrinks";
import { RumDrinksQuery } from "../api/graphql/rumDrinks";
import { TequilaDrinksQuery } from "../api/graphql/tequilaDrinks";
import { GinDrinksQuery } from "../api/graphql/ginDrinks";
import { alcoholFilterParam } from "../api/graphql/alcoholFilter";
import FilterButtons from "../components/FilterButtons";

/**
 * search filter method
 * @param cocktails all (currently shown?) cocktails
 * @param query string entered in search field
 * @returns all cocktails containing the search param
 */
const filterCocktails = (
  cocktails: {
    id: number;
    name: string;
    /*     
    description: string;
    favorite: boolean; 
    (ikke tilgjengelig i api-et, mÃ¥tte derfor fjernes)
    */
    image: string;
  }[],
  query: string
) => {
  console.log("query in filterCocktails: ", query);
  console.log("cocktails in filterCocktails: ", cocktails);
  if (query === "") {
    return cocktails;
  } else {
    const filtered = cocktails.filter((cocktail) => {
      return Object.values(cocktail.name)
        .join("")
        .toLowerCase()
        .includes(query.toLowerCase());
    });
    return filtered;
  }
};

const AllCocktailsPage = () => {
  const [query, setQuery] = useState("");
  const [cocktails, setCocktails] = useState(
    filterCocktails(dummyCocktails, query)
  );
  const filteredCocktails = filterCocktails(cocktails, query);

  return (
    <Flex flexDirection="column">
      <Header />
      <Grid
        transform="translate(0px, -150px)"
        margin={"30px"}
        rounded={20}
        templateRows="repeat(1fr)"
        templateColumns="repeat(1fr)"
        gap={4}
        mt={5}
      >
        <GridItem colSpan={3}>
          <SearchBar q={query} setQuery={setQuery} />
        </GridItem>
        <GridItem colSpan={1}>
          <FilterButtons
            setFilteredCocktails={setCocktails}
            setQuery={setQuery}
          />
        </GridItem>
        <GridItem colSpan={4}>
          <CocktailCardsDisplay cocktails={filteredCocktails} />
        </GridItem>
      </Grid>
    </Flex>
  );
};

export default AllCocktailsPage;
