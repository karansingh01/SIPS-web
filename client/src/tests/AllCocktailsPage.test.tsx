import React from "react";
import { render } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import { MockedProvider } from "@apollo/client/testing";
import { HashRouter, Route, Routes } from "react-router-dom";
import { AuthProvider } from "../context/authContext";
import client1 from "../apolloClient";
import AllCocktailsPage from "../pages/AllCocktailsPage";

test("Snapshot test of AllCocktailsPage", async () => {
  const { baseElement } = render(
    <AuthProvider>
      <ChakraProvider>
        <MockedProvider>
          <AllCocktailsPage />
        </MockedProvider>
      </ChakraProvider>
    </AuthProvider>
  );

  expect(baseElement.innerHTML).toMatchSnapshot();
});
