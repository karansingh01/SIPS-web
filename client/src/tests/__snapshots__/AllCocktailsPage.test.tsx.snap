// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of AllCocktailsPage 1`] = `"<div><p>Loading...</p><span id=\\"__chakra_env\\" hidden=\\"\\"></span></div><div class=\\"chakra-portal\\"><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-top\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;\\"></ul><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-top-left\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;\\"></ul><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-top-right\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;\\"></ul><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-bottom-left\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;\\"></ul><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-bottom\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;\\"></ul><ul role=\\"region\\" aria-live=\\"polite\\" id=\\"chakra-toast-manager-bottom-right\\" style=\\"position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;\\"></ul></div>"`;
